incremental-font-transfer/src/patchmap.rs

```diff
@@ -104,93 +105,132 @@ fn add_intersecting_format1_patches(
 
     let encoding = PatchEncoding::from_format_number(map.patch_encoding())?;
 
-    // Step 1: Collect the glyph map entries.
-    let mut entries = IntSet::<u16>::empty();
-    intersect_format1_glyph_map(font, map, codepoints, &mut entries)?;
-
-    // Step 2: Collect feature mappings.
-    intersect_format1_feature_map(map, features, &mut entries)?;
+    // Step 1: Collect the glyph and feature map entries.
+    let charmap = Charmap::new(font);
+    let mut entries = BTreeMap::<u16, SubsetDefinition>::new();
```

Comment:
not sure if this is a personal preference towards explicitness, but most of these type annotations you use on collections can be omitted.

Reply:
also, why is this mutable and passed in? Unless it's necessary for some other reason I would always prefer to have the function return a result, instead of passing in something mutable; in general I would encourage you to avoid `mut` if there's another reasonable alternative, I find it makes code much easier to understand (as in, while reading the code you don't need to keep track of whether or not something might possibly be getting mutated later, and the implications of this)